### PR TITLE
31: added headers to main page

### PR DESCRIPTION
**Pull request by anastasiianosulich**

### CODE REVIEW

File: frontend/.eslintrc
Comment (Anatolii-Ivanov) [inline frontend/.eslintrc:1]: Please, don't change these files

File: frontend/.vscode/settings.json
Comment (Anatolii-Ivanov) [inline frontend/.vscode/settings.json:1]: Please, don't change these files

File: frontend/angular.json
Comment (Anatolii-Ivanov) [inline frontend/angular.json:1]: Please, don't change these files

File: frontend/package-lock.json
Comment (Anatolii-Ivanov) [inline frontend/package-lock.json:1]: Remove it, please (from commit)

File: frontend/src/app/modules/main/main-page/headers/main-header/main-header.component.sass [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    display: flex
+    justify-content: space-between
+    align-items: center
+
+app-dropdown
+    margin-left: auto
+    margin-right: 10px
+
+.horizontal-separator
+    border-bottom: 1px solid #e8e6e6
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/main/main-page/headers/main-header/main-header.component.sass]: Colors to vars

File: frontend/src/app/modules/main/main-page/headers/navbar-header/navbar-header.component.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            <li class="navbar-item">
+                <a routerLink="./branches" routerLinkActive="active" class="navbar-link">Branches</a>
+            </li>
+            <li class="navbar-item">
+                <a routerLink="./scripts" routerLinkActive="active" class="navbar-link">Scripts</a>
+            </li>
+            <li class="navbar-item"><a routerLink="./code" routerLinkActive="active" class="navbar-link">Code</a></li>
+            <li class="navbar-item">
+                <a routerLink="./settings" routerLinkActive="active" class="navbar-link">Settings</a>
+            </li>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/main/main-page/headers/navbar-header/navbar-header.component.html]: Can you make it as a list of items, which we will load from ts file?

File: frontend/src/app/modules/main/main-page/headers/navbar-header/navbar-header.component.sass [hunk 3]
```diff
@@ -0,0 +1,53 @@
+@import "/src/styles/color"
+
+nav
+    width: 35%
+
+.navbar-list
+  list-style-type: none
+  display: flex
+  justify-content: space-around
+  padding: 1rem 0
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/main/main-page/headers/navbar-header/navbar-header.component.sass]: Let's not use rem, please, px, %, vh, vw

File: frontend/src/app/modules/main/main-routing.module.ts [hunk 4]
```diff
@@ -9,6 +9,35 @@ const routes: Routes = [
         path: '',
         component: MainComponent,
         children: [
+            {
+                path: '',
+                component: NotFoundComponent,
+                pathMatch: 'full',
+            },
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/main/main-routing.module.ts]: Why you need this one?

File: frontend/src/app/shared/components/dropdown/dropdown.component.sass [hunk 5]
```diff
@@ -0,0 +1,75 @@
+@import '../../../../styles/global'
+
+$bg-color: #ebebeb
+$shadow: rgba(0,0,0,0.05)
+$arrow-color: #8e8c8c
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/shared/components/dropdown/dropdown.component.sass]: Move to global vars (maybe we have it)

File: frontend/src/app/shared/components/dropdown/dropdown.component.sass [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    background-color: $primary-violet
+
+.dropdown.active .option
+    display: block
+
+
+
+
+
+
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/shared/components/dropdown/dropdown.component.sass]: Enters

File: frontend/src/index.html [hunk 7]
```diff
@@ -6,9 +6,12 @@
         <base href="/" />
         <meta name="viewport" content="width=device-width, initial-scale=1" />
         <link rel="icon" type="image/x-icon" href="favicon.ico" />
-    </head>
+            <link rel="preconnect" href="https://fonts.gstatic.com">
+        <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500&display=swap" rel="stylesheet">
+        <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+</head>
```
Comment (Anatolii-Ivanov) [inline frontend/src/index.html]: tabulation

File: frontend/yarn.lock
Comment (Anatolii-Ivanov) [inline frontend/yarn.lock:1]: Drop from git

File: frontend/src/styles/global.sass [hunk 8]
```diff
@@ -16,3 +18,13 @@ $app-warn: mat-palette($mat-orange, 500)
 $app-theme: mat-light-theme($app-primary, $app-accent, $app-warn)
 
 @include angular-material-theme($app-theme)
+
+
+$primary-blue: #21D7FF
+$primary-violet: #503AF9
```
Comment (Anatolii-Ivanov) [inline frontend/src/styles/global.sass]: We have separate file for colors

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]
This one is better, just some small comments :)

File: frontend/src/app/modules/main/main-page/headers/main-header/main-header.component.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+    selectedDbName: string;
+
+    dbNames: string[] = ['Branch 1', 'Branch 2', 'Branch 3', 'Branch 4'];
+
+    ngOnInit(): void {
+        this.projectName = 'Project Name';
+    }
+
+    onDatabaseSelected(value: string) {
```
Comment (OlegKarapysh) [inline frontend/src/app/modules/main/main-page/headers/main-header/main-header.component.ts]: it's better to use explicit access modifiers (public, private...) as we do in C#

File: frontend/src/app/modules/main/main-page/headers/main-header/main-header.component.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+export class MainHeaderComponent implements OnInit {
+    projectName: string;
+
+    selectedDbName: string;
+
+    dbNames: string[] = ['Branch 1', 'Branch 2', 'Branch 3', 'Branch 4'];
+
+    ngOnInit(): void {
+        this.projectName = 'Project Name';
+    }
```
Comment (OlegKarapysh) [inline frontend/src/app/modules/main/main-page/headers/main-header/main-header.component.ts]: why do you use ngOnInit here?

**Comment by anastasiianosulich** [comment]
Screenshot of the component.

Знімок екрана 2023-08-28 151456

File: frontend/src/app/modules/main/main-page/headers/navbar-header/navbar-header.component.html [hunk 11]
```diff
@@ -0,0 +1,19 @@
+<div class="container">
+    <nav>
+        <ul class="navbar-list">
+            <li *ngFor="let navLink of navLinks" class="navbar-item">
+                <a routerLink="{{ navLink.path }}" routerLinkActive="active" class="navbar-link">{{
+                    navLink.displayName
+                }}</a>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/main/main-page/headers/navbar-header/navbar-header.component.html]:
```suggestion
                <a routerLink="{{ navLink.path }}" routerLinkActive="active" class="navbar-link">
                    {{ navLink.displayName }}
                </a>
```

**Comment by OlegKarapysh** [comment]
resolve merge conflicts as quickly as possible because they reset received approvals.
do `git checkout development, git pull, git merge` before commits

File: frontend/src/app/shared/components/dropdown/dropdown.component.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+    onOptionSelected(value: string) {
+        this.selectedOption = value;
+        this.selectedValueChanged.emit(this.selectedOption);
+    }
+
+    toggleActiveClass() {
+        this.isActive = !this.isActive;
+    }
+}
```
Comment (OlegKarapysh) [inline frontend/src/app/shared/components/dropdown/dropdown.component.ts]: missing access modifiers here

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by VladTkach — APPROVED** [review]

**Review by OlegKarapysh — APPROVED** [review]

**Pull request merged**